azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java

```diff
@@ -114,8 +118,12 @@ protected void doFilterInternal(
                 authentication.setAuthenticated(true);
                 log.info("Request token verification success. {}", authentication);
                 SecurityContextHolder.getContext().setAuthentication(authentication);
-            } catch (Throwable throwable) {
-                throw new RuntimeException(throwable);
+            } catch (MalformedURLException | ParseException | BadJOSEException | JOSEException ex) {
+                log.error("Failed to initialize UserPrincipal.", ex);
+                throw new ServletException(ex);
```

Comment:
why not directly rethrow ex?

Reply:
Because 4 new exception will be thrown if throw ex here, as these 4 exception are all thrown from initializing UserPrincipal, so wrapped into 1 exception here, the original exception will still be in the exception stacktrace.